consensus/consensus-types/src/sync_info.rs

```diff
@@ -83,14 +83,16 @@ impl SyncInfo {
     }
 
     /// Highest ordered certificate
-    pub fn highest_ordered_cert(&self) -> &QuorumCert {
-        self.highest_ordered_cert
-            .as_ref()
-            .unwrap_or(&self.highest_quorum_cert)
+    pub fn highest_ordered_cert(&self) -> WrappedLedgerInfo {
+        if let Some(cert) = &self.highest_ordered_cert {
+            cert.clone()
+        } else {
+            self.highest_quorum_cert.into_wrapped_ledger_info()
```

Comment:
Why can you derive ordered proof from highest QC?

Reply:
I actually did this to imitate what is happening in the main branch already.
https://github.com/aptos-labs/aptos-core/blob/main/consensus/consensus-types/src/sync_info.rs#L86